### DISCUSSION TITLE
duplicate diagnostic item error

### DISCUSSION THREAD

**Discussion by dvic** [rust-lang/rust-analyzer#15041] [Q&A]
I'm using neovim and a regular LSP setup but I'm getting diagnostics when I go to a definition within the standard library:

```
duplicate diagnostic item in crate `core`:
`Result`.the diagnostic item is first defined in crate `core`. rustc [503, 1]
```

![image](https://github.com/rust-lang/rust-analyzer/assets/1214337/140b53a4-3b2c-4b4f-836c-307572d66d46)

I noticed also that a new LSP instance is attached when I do this (with root dir  `...../rust/library/core`). Is this supposed to happen?

Did anyone encounter this problem as well?

**Comment by lnicola** [comment]
No, I don't think it's supposed to do that. Can you try with another client?

https://helix-editor.com/ is modal and has out-of-the-box LSP support.

**Reply by dvic** [reply]
> No, I don't think it's supposed to do that. Can you try with another client?
> 
> https://helix-editor.com/ is modal and has out-of-the-box LSP support.

Hmm in helix I don't have the issue and it also seems that it's not loading a second instance of the LSP when I traverse crates (because goto is very fast, there is no indexing upon first traversal it seems)

**Comment by Veykril** [comment]
I assume neovim might report opened files as workspaces via lsp. I could imagine that causing issues like this because we then try to open the corresponding cargo project

**Reply by dvic** [reply]
> I assume neovim might report opened files as workspaces via lsp. I could imagine that causing issues like this because we then try to open the corresponding cargo project

If that's the case, is there a way to override the "root dir"? Or is this a setting on the neovim lsp side of things?

**Comment by dvic** [comment]
I think I found a clue in the nvim-lspconfig repo: https://github.com/neovim/nvim-lspconfig/blob/5da57456597ac2f25f20d6ecfae7b7c9cbd01fc6/lua/lspconfig/server_configurations/rust_analyzer.lua#L41-L44

It looks like there is custom logic for reusing an instance...
